robot/pmm/tests/browser/ProgramEngagement/quick_actions_sd.robot

```diff
@@ -1,50 +1,65 @@
+...             robot/pmm/resources/ServiceDeliveryPageObject.py
 Suite Setup     Run Keywords
 ...             Open Test Browser
 ...             Setup Test Data
-Suite Teardown  Delete Records and Close Browser
-
+Suite Teardown  Capture Screenshot and Delete Records and Close Browser
 *** Keywords ***
 Setup Test Data
     ${ns} =                     Get PMM Namespace Prefix
     Set suite variable          ${ns}
+
     &{program} =                API Create Program
-    Store Session Record        ${ns}Program__c                 &{program}[Id]
     Set suite variable          &{program}
     &{contact} =                API Create Contact
     Set suite variable          &{contact}
-    &{program_engagement} =     API Create Program Engagement   &{Program}[Id]      &{contact}[Id]
+    &{contact1} =               API Create Contact
+    Set suite variable          &{contact1}
+    &{program_engagement} =     API Create Program Engagement   ${Program}[Id]      ${contact}[Id]
     Set suite variable          &{program_engagement}
-    &{service} =                API Create Service              &{Program}[Id]
+    &{service} =                API Create Service              ${Program}[Id]
     Set suite variable          &{service}
+
     ${today} =                  Get Current Date                result_format=%Y-%m-%d
     Set suite variable          ${today}
     ${quantity} =               Generate Random String    2     [NUMBERS]
     Set suite variable          ${quantity}
 
 *** Test Cases ***
 Create a new service delivery using quick action
+     [Documentation]                This test loads the program engagement record, clicks on the new service delivery quick action and creates
+     ...                            new service delivery record.
+     [tags]                         W-037572  feature:Service Delivery
      Go To PMM App
-     Go To Page                  Details              ProgramEngagement__c                   object_id=&{program_engagement}[Id]
-     page should contain         &{program_engagement}[Name]
-     click quick action button   Create New Service Delivery
-     Load Page Object            NewServiceDelivery   ${ns}ServiceDelivery__c
-     verify current page title   Create New Service Delivery
-     populate modal form         Client=&{contact}[FirstName] &{contact}[LastName]
-     ...                         Service=&{service}[Name]
-     ...                         Quantity=${quantity}
-     Click Modal button          Save
+     Go To Page                     Details                        ProgramEngagement__c      object_id=${program_engagement}[Id]
+     Verify Details                 Program Engagement Name        contains                       ${contact}[FirstName] ${contact}[LastName] ${today}: ${program}[Name]
+     Click Quick Action Button      Create New Service Delivery
+     Verify Current Page Title      Create New Service Delivery
```

Comment:
can this verify step be added as part of click quick action button?

Reply:
Fixed